ut/columns_ut.cpp

```diff
@@ -114,6 +152,135 @@ TEST(ColumnsCase, DateAppend) {
+
+    const auto data = MakeDateTime64s();
+    for (const auto & v : data) {
+        column->Append(v);
+    }
+
+    ASSERT_EQ(data.size(), column->Size());
+    for (size_t i = 0; i < data.size(); ++i) {
+        ASSERT_EQ(data[i], column->At(i));
+    }
+}
+
+TEST(ColumnsCase, DateTime64_Clear) {
+    auto column = std::make_shared<ColumnDateTime64>(6ul);
+
+    // Clearing empty column doesn't crash and produces expected result
+    ASSERT_NO_THROW(column->Clear());
+    ASSERT_EQ(0u, column->Size());
+
+    const auto data = MakeDateTime64s();
+    for (const auto & v : data) {
+        column->Append(v);
+    }
+
+    ASSERT_NO_THROW(column->Clear());
+    ASSERT_EQ(0u, column->Size());
+}
+
+TEST(ColumnsCase, DateTime64_Swap) {
+    auto column = std::make_shared<ColumnDateTime64>(6ul);
+
+    const auto data = MakeDateTime64s();
+    for (const auto & v : data) {
+        column->Append(v);
+    }
+
+    auto column2 = std::make_shared<ColumnDateTime64>(6ul);
+    column->Swap(*column2);
+
+    // Validate that all items were transferred to column2.
+    ASSERT_EQ(0u, column->Size());
+    ASSERT_EQ(data.size(), column2->Size());
+
+    // Clearing doesn't affects precision value.
+    ASSERT_EQ(6ul, column->GetPrecision());
+
+    for (size_t i = 0; i < data.size(); ++i) {
+        ASSERT_EQ(data[i], column2->At(i));
+    }
+}
+
+TEST(ColumnsCase, DateTime64_Slice) {
+    auto column = std::make_shared<ColumnDateTime64>(6ul);
+
+    {
+        // Empty slice on empty column
+        auto slice = column->Slice(0, 0)->As<ColumnDateTime64>();
+        ASSERT_EQ(0u, slice->Size());
+        ASSERT_EQ(column->GetPrecision(), slice->GetPrecision());
+    }
+
+    const auto data = MakeDateTime64s();
+    const size_t size = data.size();
+    ASSERT_GT(size, 4u); // so the partial slice below has half of the elements of the column
+
+    for (const auto & v : data) {
+        column->Append(v);
+    }
+
+    {
+        // Empty slice on non-empty column
+        auto slice = column->Slice(0, 0)->As<ColumnDateTime64>();
+        ASSERT_EQ(0u, slice->Size());
+        ASSERT_EQ(column->GetPrecision(), slice->GetPrecision());
+    }
+
+    {
+        // Full-slice on non-empty column
+        auto slice = column->Slice(0, size)->As<ColumnDateTime64>();
+        ASSERT_EQ(column->Size(), slice->Size());
+        ASSERT_EQ(column->GetPrecision(), slice->GetPrecision());
+
+        for (size_t i = 0; i < data.size(); ++i) {
+            ASSERT_EQ(data[i], slice->At(i));
+        }
+    }
+
+    {
+        const size_t offset = size / 4;
+        const size_t count = size / 2;
+        // Partial slice on non-empty column
+        auto slice = column->Slice(offset, count)->As<ColumnDateTime64>();
+
+        ASSERT_EQ(count, slice->Size());
+        ASSERT_EQ(column->GetPrecision(), slice->GetPrecision());
+
+        for (size_t i = offset; i < offset + count; ++i) {
+            ASSERT_EQ(data[i], slice->At(i - offset));
+        }
+    }
+}
```

Comment:
What happens on out-of-bound slices (5 cases)?

Reply:
Added tests